tests/jballerina-unit-test/src/test/resources/test-src/javainterop/basic/variable_number_of_resource_paths.bal

```diff
@@ -0,0 +1,91 @@
+// specific language governing permissions and limitations
+// under the License.
+
+import ballerina/jballerina.java;
+import ballerina/test;
+
+public client isolated class Client {
+
+    public isolated function init() {
+
+    }
+
+    isolated resource function get abc/[string ...path] () returns int = @java:Method {
+        'class: "org/ballerinalang/nativeimpl/jvm/tests/StaticMethods",
+        name: "getResource",
+        paramTypes: ["io.ballerina.runtime.api.values.BObject", "io.ballerina.runtime.api.values.BArray"]
+    } external;
+
+    isolated resource function get abc/[int id]/[float v]/[string path]/[string ...paths] () returns int = @java:Method {
+        'class: "org/ballerinalang/nativeimpl/jvm/tests/StaticMethods",
+        name: "getResource",
+        paramTypes: ["io.ballerina.runtime.api.values.BObject", "io.ballerina.runtime.api.values.BArray"]
+    } external;
+
+    isolated resource function get def/[string path]/[int path1]/[string path2]/[int ...path3] () returns int = @java:Method {
+        'class: "org/ballerinalang/nativeimpl/jvm/tests/StaticMethods",
+        name: "getResource",
+        paramTypes: ["io.ballerina.runtime.api.values.BObject", "io.ballerina.runtime.api.values.BArray"]
+    } external;
+
+    isolated resource function get abc/[int id]/[string path] (float f, string s) returns int = @java:Method {
+        'class: "org/ballerinalang/nativeimpl/jvm/tests/StaticMethods",
+        name: "getResource",
+        paramTypes: ["io.ballerina.runtime.api.values.BObject", "io.ballerina.runtime.api.values.BArray", "double", "io.ballerina.runtime.api.values.BString"]
+    } external;
+
+    isolated resource function get abc/[int id]/[string path]/[string p2] (string a, float f, string s) returns int = @java:Method {
+        'class: "org/ballerinalang/nativeimpl/jvm/tests/StaticMethods",
+        name: "getResource",
+        paramTypes: ["io.ballerina.runtime.api.values.BObject", "io.ballerina.runtime.api.values.BArray", "io.ballerina.runtime.api.values.BString", "double", "io.ballerina.runtime.api.values.BString"]
+    } external;
+
+    isolated resource function get def/[int id]/[string path]/[string p2] (string s) returns string = @java:Method {
+        'class: "org/ballerinalang/nativeimpl/jvm/tests/StaticMethods",
+        name: "getResource",
+        paramTypes: ["io.ballerina.runtime.api.Environment", "io.ballerina.runtime.api.values.BObject", "io.ballerina.runtime.api.values.BArray", "io.ballerina.runtime.api.values.BString"]
+    } external;
+
+    isolated resource function get ghi/[int id]/[string p1]/[string p2]/[int ...ids] (string s) returns string = @java:Method {
+        'class: "org/ballerinalang/nativeimpl/jvm/tests/StaticMethods",
+        name: "getResource",
+        paramTypes: ["io.ballerina.runtime.api.Environment", "io.ballerina.runtime.api.values.BObject", "io.ballerina.runtime.api.values.BArray", "io.ballerina.runtime.api.values.BString"]
```

Comment:
Shall we add a test without param types and with only one function with given name.